README.md

```diff
@@ -21,13 +21,91 @@ Then generate your new project:
 yo node-graphql
 ```
 
+## Generated code
+
+If all default options are entered, the code generated should include a few runnable tests, as well as a runnable server.
+
+### Tests
+
+The generated tests are written using `Jest`, and provide an in memory database using `mongodb-memory-server`, as well as means to test any graphql resolvers and schema validations using `easygraphql-tester`.
+
+The tests can be run as soon as the code is generated with:
+
+```bash
+npm run test
+```
+
+### Source Code
+
+The source code uses `apollo-server-express` along with `express` for graphql and the server, with the `jsonwebtoken` package to handle JWTs. It also uses `lodash`, `merge-graphql-schemas` and `graphql-import-node` for ease of use, as well as `dotenv` for its environment files.
+
+The server can be run with nodemon by:
+
+```bash
+npm run dev
+```
+
+### Database
+
+By default (and currently only choice), the chosen database is noSQL, which uses the `mongoose` package. While the tests can run fine without any setup here, to properly run the server, an instance of `mongoDB` should be created for the system to connect to. Please check the `.env` file and place your database credentials there.
+
+### What the code can do
+
+Out of the box, if you have already set up a database, run the server and navigate to `localhost:<PORT>/graphql`
+
+Out of the box, the system provides a method of creating a user, logging in, and checking the logged in user, all of this, using JWT for auth. If you navigate to the `user.graphql` file, you can see the provided mutations. Feel free to extend this here for your own system. But for now, if we want to create a user, we can simply enter the following in graphql playground
+
+```gql
+mutation {
+  signup(username: "user", email: "test@email.com", password: "password")
+}
+```
+
+Which will create a new user and return us a token.
+
+From here, we can either use that token, or test the login, which we can do as follows
+
+```gql
+mutation {
+  login(email: "test@email.com", password: "password")
+}
+```
+
+Which also returns us a token. Now lets try check who is currently logged in with our token.
```

Comment:
"Which also returns us a token. Now lets try check who is currently logged in with our token." => "The above also returns us a `<type of token (refresh. etc)>` token. So now, let's verify and check who is currently logged in with our token."